### PR TITLE
Integration & Structure testing

### PR DESCRIPTION
**Pull request by aslo**
Adds structure tests using the runtimes_common framework, and integration tests using a custom framework.  This PR is mostly focused on the frameworks, so the number of test cases are relatively light.

## Integration tests
Run all integration tests manually using `tests/integration/test-suite.sh`. Normally, this would be called from the `ci-build.sh` script (which is run by our CI system whenever new code is pushed to master).

Each test is a subdirectory of `tests/integration/test_cases/`. For each test case, the test framework invokes the builder pipeline on the directory defined in the test's `repo.cfg` file (relative to the `test/integration/test_apps`). A set of structure tests are then run on the resulting application container.  

## Structure tests
Structure tests are just run normally as part of the regular cloud build of the runtime-builder image.

### CODE REVIEW

File: test/integration/test_cases/petclinic/repo.cfg [hunk 1]
```diff
@@ -0,0 +1 @@
+GIT_REPO="git@github.com:spring-projects/spring-petclinic.git"
```
Comment (meltsufin) [inline test/integration/test_cases/petclinic/repo.cfg]: Let's avoid third-party repos.

File: test/integration/test_cases/helloworld-war/repo.cfg [hunk 2]
```diff
@@ -0,0 +1 @@
+GIT_REPO="https://github.com/GoogleCloudPlatform/java-docs-samples.git"
```
Comment (meltsufin) [inline test/integration/test_cases/helloworld-war/repo.cfg]: Checkout specific commit for repeatable builds.

**Comment by aslo** [comment]
@meltsufin PTAL - I updated the integration test framework to be more flexible WRT test apps. Now, any subdirectory of `test/integration/test_apps` can be used as test fixtures. These test fixtures can either be defined locally, or references to external repos (via git submodules).

File: test/integration/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Integration Testing
```
Comment (meltsufin) [inline test/integration/README.md]: Can you link to this readme from DEVELOPING.md? Also, the explanation of when the tests run etc., can go into the DEVELOPING.md.

Comment (meltsufin) [inline test/integration/README.md]: ...the stuff you wrote in the PR description

**Pull request merged**